### ISSUE TITLE
A11y: [calcite-tree] Can't navigate through all tree items using arrow keys

### ISSUE THREAD

**Issue by ellenupp** [Esri/calcite-components#3123]
### Summary
The arrow key navigation in `calcite-tree` does not function as expected and it is not always possible to reach every item using the keyboard alone.

### Actual Behavior
Left arrow key: works as expected
Right arrow key: with focus on expanded parent, right arrow moves focus up instead of to first child
Up/down arrow keys: work fine for first level of tree, but when focus is on a deeper level item, using up/down moves the focus back out to the top level instead of to the element's next sibling. See repro steps for example. 

### Expected Behavior
All items in the tree should be reachable using the keyboard. Up and down arrow keys should move focus between adjacent tree items. Right arrow key should expand a parent container if collapsed, else move focus to its first child. W3C links below for examples. Please contact me for link to a custom tree implementation in dashboards for another example of expected behaviour for navigating between tree items. 

### Reproduction Steps

Code pen sample: https://codepen.io/ellupp/pen/PojWvmR?editors=1100
1. Click tree item "Rich text 2"
2. Press up arrow key. Focus jumps out two levels instead of moving up to previous sibling "Rich text 1".
2. Click tree item "Rich text 1". Press down arrow key. Focus jumps out of current level and down to parent's sibling instead of going to that item's next sibling "Rich text 2".
3. Press right arrow key (with focus now on last "Column"). Focus jumps up to element's first sibling instead of to its first child "Serial chart". It doesn't seem possible to navigate to the children. 

### Working W3C Example/Tutorial 
https://w3c.github.io/aria-practices/#TreeView
https://www.w3.org/TR/wai-aria-practices-1.1/examples/treeview/treeview-2/treeview-2a.html

cc @vijayendranj @julio8a

**Comment by caripizza** [comment]
> Please contact me for link to a custom tree implementation in dashboards for another example of expected behaviour for navigating between tree items.

Thanks for adding this issue @ellenupp, can you please share this link with me when you have a moment?

**Comment by Elijbet** [comment]
It seems like the navigation doesn’t work because it’s handled on the tree item level and is unaware that the current state of the entire tree (and of the nested tree items). It moves up/down siblings only. Looks like the whole keyboard interaction logic needs to be moved a level up so the arrow up/down can move to next node whether it’s a sibling, a parent, or a child. Also so that I can have a property that checks what I’m focused on currently, which will allow me to check if the item is expandable or not so I can figure out the right arrow behavior based on that. 

@driskull @jcfranco

**Comment by driskull** [comment]
I agree with that assessment @Elijbet. The parent should handle the interaction. This way, there are only listeners setup on the parent component which is a good refactor.

**Comment by benelan** [comment]
verified on beta.79 🎉
